### ISSUE TITLE
Remove tls mode

### ISSUE THREAD

**Issue by Neilpang** [acmesh-official/acme.sh#1322]
tls mode is removed from ACME protocol: 
https://github.com/letsencrypt/boulder/issues/3297#issuecomment-369939882

we will remove it too.

**Comment by FernandoMiguel** [comment]
expected.
thanks for the heads up

**Comment by cpu** [comment]
Note that while TLS-SNI-01 is removed from the specification Boulder and Let's Encrypt's V1 API still support TLS-SNI-01 for the [two cases we described here](https://community.letsencrypt.org/t/tls-sni-challenges-disabled-for-most-new-issuance/50316). We'll eventually remove support from Boulder entirely but it hasn't happened yet and we haven't picked a fixed date to do so. You might want to wait for Boulder to remove the code before you delete it yourself (or not! up to you ;-))

**Comment by Neilpang** [comment]
@cpu   thank you.
We will wait for a period.

**Comment by Neilpang** [comment]
Let's remove the  tls doc first.

**Comment by xj25vm** [comment]
It would be really nice if the --tls would have been left in acme.sh, and would just return an error similar to:

"This option has been deprecated at Let's Encrypt and removed from acme.sh"

When I tried to use it recently it just returns "Unknown parameter --tls" - which is really confusing, as it used to be an option until recently, and I banged by my head against it for a while, thinking that I copied my scripts and command wrong somehow. It isn't even mentioned on the https://github.com/Neilpang/acme.sh/wiki/How-to-issue-a-cert page - so it took me a while to find this thread. It would be user friendly if it would be mentioned in the docs that the --tls has been removed. At the moment all mentions of it seem to have completely vanished.
